packages/base/src/trace-manager.ts

```diff
@@ -1,5 +1,6 @@
 
 import { Trace } from 'tsp-typescript-client/lib/models/trace';
+import { Experiment } from 'tsp-typescript-client/lib/models/experiment';
```

Comment:
this adds a circular dependency between experiment manager and trace manager. Could you listen to the signal in the Experiment manager and then call the delete trace on the trace manager instead?

Reply:
Actually it's only a dependency on the experiment model, not the experiment manager. Is it OK?

I couldn't do it in the experiment manager because it does not have any way to get a reference to the trace manager (that I know of)

Reply:
I had the impression that experiment manager has already access to Trace Manager. I wonder what makes more sense, trace manager references experiment manager or vice versa. Same question is about experiment model reference to trace model or vice versa.

To get access to the trace manager you could do:
@inject(TspClientProvider) private tspClientProvider: TspClientProvider
and then:  tspClientProvider.getTraceManager().

Reply:
I realized that using dependency injection would create a dependency to viewer-prototype and hence Theia which would make it difficult to migrate to VsCode.